### PR TITLE
team mentions basic

### PR DESCRIPTION
**Pull request by mmaxim**
Team mentions in chat, allowing you to @ mention a team and see a hover box.

**Go**
1. Add `TeamMentionLoader` to allow for async loading of mention info from the server. Just sits in a loop hitting the new server endpoint.
2. Change @ mention parsing to include names with '.''s and the @keybase#general form.
3. Consider any @ mention that isn't a user a possible team mention, and queue up the load on `TeamMentionLoader` to be sent to the frontend async.

**JS**
1. Factor out the `TeamInfo` component from profile so we can use it in other places.
2. Add the `TeamMention` component to be used in chat messages through the service decoration markdown component.
3. Add a section in the store for team name -> description info.

### CODE REVIEW

**Comment by mmaxim** [comment]
cc @malgorithms

File: go/chat/utils/utils.go [hunk 1]
```diff
@@ -610,21 +611,23 @@ func parseRegexpNames(ctx context.Context, body string, re *regexp.Regexp) (res
 	allIndexMatches := re.FindAllStringSubmatchIndex(body, -1)
 	for _, indexMatch := range allIndexMatches {
 		if len(indexMatch) >= 4 {
-			hit := body[indexMatch[2]:indexMatch[3]]
+			low := indexMatch[2] + 1
```
Comment (joshblum) [inline go/chat/utils/utils.go]: can you add a comment here for the `+1` makes it easier to follow

File: go/service/main.go [hunk 2]
```diff
@@ -399,6 +399,7 @@ func (d *Service) startChatModules() {
 		g.Indexer.Start(globals.ChatCtx(context.Background(), g,
 			keybase1.TLFIdentifyBehavior_CHAT_SKIP, nil, nil), uid)
 		g.CoinFlipManager.Start(context.Background(), uid)
+		g.TeamMentionLoader.Start(context.Background(), uid)
```
Comment (joshblum) [inline go/service/main.go]: also add to `kbtest/chat.go`?

File: go/chat/utils/utils.go [hunk 3]
```diff
@@ -1993,30 +2021,46 @@ func DecorateBody(ctx context.Context, body string, offset, length int, decorati
...
+		for _, tm := range teamMentions {
+			name := tm.Name
+			if len(tm.Channel) > 0 {
+				name += "#" + tm.Channel
+			}
+			teamMap[name] = tm
+		}
 		for _, m := range atMatches {
-			switch m.name {
-			case "here", "channel", "everyone":
```
Comment (joshblum) [inline go/chat/utils/utils.go]: prefer this style ¯\_(ツ)_/¯

File: go/service/main.go (hunk 2 above)
Comment (mmaxim) [reply go/service/main.go]: I'm going to add a test

File: shared/actions/chat2/index.js [hunk 4]
```diff
@@ -3139,6 +3147,10 @@ function* chat2Saga(): Saga.SagaGenerator<any, any> {
     EngineGen.chat1ChatUiChatCommandMarkdown,
     onChatCommandMarkdown
   )
+  yield* Saga.chainAction<EngineGen.Chat1ChatUiChatTeamMentionUpdatePayload>(
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: you can just handle this in a reducer and not make a middle man action just fyi

File: shared/chat/conversation/team-mention/container.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import TeamMention from '.'
+
+type OwnProps = {|
+  allowFontScaling?: boolean,
+  channel: string,
+  name: string,
+  style?: Styles.StylesCrossPlatform,
+|}
+
+const mapStateToProps = (state, {allowFontScaling, name, channel, style}: OwnProps) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/team-mention/container.js]: you don't need to do `:OwnProps` here

File: shared/chat/conversation/team-mention/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
+// @flow
+import React from 'react'
+import Text from '../../../common-adapters/text'
+import {Box2} from '../../../common-adapters/box'
+import * as Styles from '../../../styles'
+import TeamInfo from '../../../profile/user/teams/teaminfo'
```
Comment (chrisnojima) [inline shared/chat/conversation/team-mention/index.js]: we now do 
```
const Kb = {Box2, Text}
```
in files that can't import * common so the jsx looks the same (`Kb.Box2`) etc

File: shared/chat/conversation/team-mention/index.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  isOpen: boolean,
+  name: string,
+  onJoinTeam: string => void,
+  resolved: boolean,
+  numMembers: number,
+  publicAdmins: Array<string>,
+  style?: Styles.StylesCrossPlatform,
+|}
+
+type State = {
```
Comment (chrisnojima) [inline shared/chat/conversation/team-mention/index.js]: exact

**Pull request merged**